data-structures/map-query.ts

```diff
@@ -0,0 +1,16 @@
+import { Document, DocumentIndex } from './map-index';
+
+/**
+ * Queries a DocumentIndex and returns all the Documents that contain
+ * any of the words in the query, as a Set.
+ */
+export function queryIndex(index: DocumentIndex, query: string): Set<Document> {
+  let set = new Set<Document>();
+  const words = query.toLowerCase().match(/\b(\w+)\b/g);
+  if (!words) return set;
```

Comment:
It looks like the code will do the right thing if you leave off this check, because the for-loop at line 11 will not execute anything.